test/index.js

```diff
@@ -196,221 +365,103 @@ orchestrator.registerScenario(
-      'add_vote_of_goal',
-      {
-        goal_vote: {
-          goal_address: goal.Ok.goal.address,
-          urgency: 0.5,
-          importance: 0.5,
-          impact: 0.5,
-          effort: 0.5,
-          agent_address: alice._instances.acorn_hc.agentAddress,
-          unix_timestamp: Date.now(),
-        },
-      }
-    )
-    await s.consistency()
-
-    const result_alex2 = await alex.call(
-      'acorn_hc',
-      'holo_acorn',
-      'fetch_goal_members',
-      {}
-    )
-    const result_alex5 = await alex.call(
-      'acorn_hc',
-      'holo_acorn',
-      'fetch_goal_votes',
-      {}
-    )
-    const result_alice = await alice.call(
-      'acorn_hc',
-      'holo_acorn',
-      'fetch_goals',
-      {}
-    )
-    t.isNotEqual(goal2.Ok.maybe_edge, null)
-    t.equal(result_alice.Ok.length, 2)
-    t.deepEqual(result_bob.Ok.entry, {
-      content: 'sample content2',
-      user_hash: alice._instances.acorn_hc.agentAddress,
-      unix_timestamp: time,
-      hierarchy: 'Root',
-      status: 'Uncertain',
-      description: '33',
+      address: create_goal.Ok.goal.address,
     })
-    t.deepEqual(result_alex.Ok.entry, result_alex2.Ok[0].entry)
-    t.deepEqual(result_alex4.Ok.entry, result_alex5.Ok[0].entry)
-    const result_alex7 = await alex.call(
-      'acorn_hc',
-      'holo_acorn',
-      'update_goal_vote',
-      {
-        goal_vote: {
-          goal_address: goal.Ok.goal.address,
-          urgency: 0,
-          importance: 0,
-          impact: 0,
-          effort: 0,
-          agent_address: alice._instances.acorn_hc.agentAddress,
-          unix_timestamp: Date.now(),
-        },
-        address: result_alex4.Ok.address,
-      }
-    )
     await s.consistency()
-    const result_alex8 = await alex.call(
-      'acorn_hc',
-      'holo_acorn',
-      'fetch_goal_votes',
-      {}
-    )
-
-    t.deepEqual(result_alex7.Ok.entry, result_alex8.Ok[0].entry)
-    const result_alice2 = await alice.call(
-      'acorn_hc',
-      'holo_acorn',
-      'archive_goal',
-      { address: goal.Ok.goal.address }
-    )
-    await s.consistency()
-    const result_alice3 = await alice.call(
-      'acorn_hc',
-      'holo_acorn',
-      'fetch_edges',
-      {}
-    )
-    const result_bob2 = await bob.call(
+    const second_fetch_goals_result = await alice.call(
       'acorn_hc',
       'holo_acorn',
       'fetch_goals',
       {}
     )
-
-    const result_alex3 = await alex.call(
-      'acorn_hc',
-      'holo_acorn',
-      'fetch_goal_members',
-      {}
-    )
-    const result_alex6 = await alex.call(
-      'acorn_hc',
-      'holo_acorn',
-      'fetch_goal_votes',
-      {}
-    )
-
-    t.equal(result_alice2.Ok.address, goal.Ok.goal.address)
-    t.equal(result_alice3.Ok.length, 0)
-    t.equal(result_alex3.Ok.length, 0)
-    t.equal(result_alex6.Ok.length, 0)
-
-    t.equal(result_bob2.Ok.length, 1)
+    t.equal(first_fetch_goals_result.Ok[0].address, second_fetch_goals_result.Ok[0].address)
```

Comment:
@hector0513 I've added the test, and commented out the rest (for now). 

If you get this test passing, that will mean we've achieved the goal. Of course, this test would already be passing on `master` branch, but the goal is to improve that code.